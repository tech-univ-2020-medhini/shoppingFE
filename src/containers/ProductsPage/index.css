.product-page{
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-around;
    padding: 1rem;
    margin: 1rem;
}