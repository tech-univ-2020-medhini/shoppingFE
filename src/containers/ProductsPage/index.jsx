import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import ProductList from '../ProductsList';
import FilterBar from '../../components/FilterBar';
import './index.css';

const ProductsPage = (props) => {
  const { productList } = props;
  const [categories, setCategories] = useState([]);
  const [selectedCategory, setSelectedCategory] = useState('All');
  useEffect(() => {
    const getCategories = async () => {
      const response = await axios.get('http://localhost:8080/categories');
      setCategories(response.data);
    };
    getCategories();
  }, []);
  return (
    <div className="products-page">
      <div>Filter categories</div>
      <FilterBar categoryList={categories} onCLickFilter={setSelectedCategory} />
      <ProductList filterCategory={selectedCategory} productList={productList} />
    </div>
  );
};
ProductsPage.propTypes = {
  productList: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default ProductsPage;
