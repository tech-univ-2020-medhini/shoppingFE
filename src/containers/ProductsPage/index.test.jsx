import React from 'react';
import { render } from '@testing-library/react';
import axios from 'axios';
import ProductsPage from './index';


describe('the products page container', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<ProductsPage />);

    expect(asFragment()).toMatchSnapshot();
  });
  it('should make an api call when rendered', () => {
    const mockAxios = jest.spyOn(axios, 'get');
    mockAxios.mockResolvedValue({ data: [] });

    render(<ProductsPage />);
    expect(mockAxios).toHaveBeenCalledWith('http://localhost:8080/products');
  });
});
