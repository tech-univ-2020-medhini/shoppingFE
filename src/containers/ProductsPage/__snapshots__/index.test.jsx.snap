// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the products page container should render correctly 1`] = `
<DocumentFragment>
  <div
    class="products-page"
  >
    <div
      class="products-header"
    >
      <div
        class="product-page-text"
      >
        Filter categories
      </div>
      <div
        class="filter-bar"
      >
        <button
          class="category-button"
          type="button"
        >
          All
        </button>
      </div>
    </div>
    <div
      class="products-container"
    >
      <p>
        All
      </p>
      <div
        class="products-list"
      />
    </div>
  </div>
</DocumentFragment>
`;
